src/main/flight/alt_hold.c

```diff
@@ -90,65 +89,66 @@ void altHoldProcessTransitions(void) {
+    // User can adjust the target altitude with throttle, but only when
+    // - throttle is outside deadband, and 
+    // - throttle is not low (zero), and 
+    // - deadband is not configured to zero
+
+    float stickFactor = 0.0f;
+
+    if (altHold.allowStickAdjustment && calculateThrottleStatus() != THROTTLE_LOW) {
+        const float rcThrottle = rcCommand[THROTTLE];
+        const float lowThreshold = autopilotConfig()->hover_throttle - altHold.deadband * (autopilotConfig()->hover_throttle - PWM_RANGE_MIN);
+        const float highThreshold = autopilotConfig()->hover_throttle + altHold.deadband * (PWM_RANGE_MAX - autopilotConfig()->hover_throttle);
+
+        if (rcThrottle < lowThreshold) {
+            stickFactor = scaleRangef(rcThrottle, PWM_RANGE_MIN, lowThreshold, -1.0f, 0.0f);
+        } else if (rcThrottle > highThreshold) {
+            stickFactor = scaleRangef(rcThrottle, highThreshold, PWM_RANGE_MAX, 0.0f, 1.0f);
+        }
     }
-    // if failsafe is active, and we get here, we are in failsafe landing mode
+
+    // if failsafe is active, and we get here, we are in failsafe landing mode, it controls throttle
     if (failsafeIsActive()) {
         // descend at up to 10 times faster when high
-        // default landing time is now 60s; need to get the quad down in this time from reasonable height
-        // need a rapid descent if high to avoid that timeout, and must slow down closer to ground
+        // default landing timeout is now 60s; must to get the quad down within this limit
+        // need a rapid descent when initiated high, and must slow down closer to ground
         // this code doubles descent rate at 20m, to max 10x (10m/s on defaults) at 200m
-        // user should be able to descend within 60s from around 150m high without disarming, on defaults
         // the deceleration may be a bit rocky if it starts very high up
         // constant (set) deceleration target in the last 2m
-        throttleAdjustmentFactor = -(0.9f + constrainf(getAltitudeCm() / 2000.0f, 0.1f, 9.0f));
+        stickFactor = -(0.9f + constrainf(getAltitudeCm() / 2000.0f, 0.1f, 9.0f));
     }
-    altHoldState.targetAltitudeAdjustRate = throttleAdjustmentFactor * altholdConfig()->alt_hold_target_adjust_rate;
+    altHold.targetVelocity = stickFactor * altHold.maxVelocity;
 
-    // if taskRate is 100Hz, default adjustRate of 100 adds/subtracts 1m every second, or 1cm per task run, at full stick position
-    altHoldState.targetAltitudeCm += altHoldState.targetAltitudeAdjustRate  * taskIntervalSeconds;
+    // prevent stick input from moving target altitude too far away from current altitude
+    // otherwise it can be difficult to bring target to the other side of current in reasonable time
+    // this constrains the P and I response to user target changes, but not D of F responses
+    if (fabsf(getAltitudeCm() - altHold.targetAltitudeCm) < altHold.maxVelocity) {
```

Comment:
Good point - I'm mixing velocity and distance.
Using the configured velocity value stops the pilot moving the target altitude more than 1s of stick adjustment away from the current altitude.
If the pilot is monitoring the current altitude, it is otherwise possible to hold throttle down in 'rising air lift' or after 'high speed ascents' and push target so far below current altitude that the subsequent drop is well below where the altitude at which the pilot thought it should stop.  Most pilots are not accustomed to the idea that the target altitude will change all the time when the sticks are outside deadband.
In essence this approach means that regardless of the pilot's configured max ascend/descend rate, they cannot force the target altitude so far away that they cannot return target to current within one second.

Reply:
The algorithm is fine, my problem is that 1s constant used is invisible in the code.

Reply:
I see.  Added your suggestion and a comment:
// Range is compared to distance that might be traveled in one second